src/main/resources/org/jenkinsci/plugins/authorizeproject/strategy/SpecificUsersAuthorizationStrategy/checkPasswordRequested.js

```diff
@@ -45,16 +45,16 @@ Behaviour.specify(".specific-user-authorization", "checkPasswordRequired", 0, fu
   
   var useridField = findFormItem(e, "userid", findChild(e));
   var passwordField = findFormItem(e, "password", findChild(e));
-  var passwordFieldBlock = findAncestor(passwordField, "TR");
+  var passwordFieldBlock = passwordField.closest("TR");
   /*
     [JENKINS-64341] - In Jenkins version greater than 2.263.4
     the table tags for fields (in forms) were replaced by div tags.
-    Since the changes were applied the 'findAncestor(passwordField, "TR")'
+    Since the changes were applied the 'passwordField.closest("TR")'
     method returns null. It is necessary to use the method
-    'findAncestorClass(passwordField, "tr")' instead of.
+    'passwordField.closest(".tr")' instead.

```

Comment:
I don't understand the extra "." that is added before the tag.  The [new code](https://github.com/jenkinsci/jenkins/blob/551043065bc672102fb8b07e8008f959ebc566fa/war/src/main/webapp/scripts/hudson-behavior.js#L456) seems to already add a leading ".".  Is there  a reason we need two dots?
```suggestion
    'passwordField.closest("tr")' instead.

```

Reply:
The new code is not called from here, instead we're calling `closest` directly so we have to add the dot ourselves.

`e.closest("tr")` has the same semantic as the deprected `findAncestor(e,"tr")`
`e.closest(".tr")` has the same semantic as the deprected `findAncestorClass(e,"tr")`

It's also questionable if using both `e.closest("TR")` and `e.closest(".tr")` is still needed, given that required core version is 2.387 now which includes the tables-to-divs migration already.

Reply:
Thanks very much.  I should have read the deprecated method more carefully.  Your description makes it clear.  Thanks again